src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -115,7 +115,7 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
         if (containingType.IsRefLikeType)
         {
             statements.Add(factory.ReturnStatement(factory.FalseLiteralExpression()));
-            return statements.ToImmutable();
+            return statements.ToImmutableAndClear();
```

Comment:
>return statements.ToImmutableAndClear(); [](http://example.com/codeflow?start=12&length=40)

nit: could just use return a collection expression in this condition and move the arraybuilder decl to after the condition